### PR TITLE
Basic framework code

### PR DESCRIPTION
**Pull request by sbryzak**
Implement components for simple framework in order to cleanly separate controller / model layers

Fixes https://issues.redhat.com/browse/CRT-1625

### CODE REVIEW

File: pkg/application/factory/manager/manager.go [hunk 1]
```diff
@@ -1,15 +1,17 @@
 package factory
 
+import (
+	contextPkg "github.com/codeready-toolchain/sandbox-auth/pkg/application/service/context"
```
Comment (alexeykazakov) [inline pkg/application/factory/manager/manager.go]: Let's not use camelCase for package names as the golang convention recommends: https://go.dev/blog/package-names

File: pkg/application/service/base/base_service.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package base
+
+import (
+	contextPkg "github.com/codeready-toolchain/sandbox-auth/pkg/application/service/context"
```
Comment (alexeykazakov) [inline pkg/application/service/base/base_service.go]: Same here.

**Review by alexeykazakov — APPROVED** [review]

**Comment by xcoulon** [comment]
Why do we need this? Can't we just have services and interfaces so we can mock them in unit tests? 
I'm worried we're heading towards an unnecessary complex framework.

I believe we could start with something like this:
```

type ServiceA interface {
  DoSomething()
  DoSomethingElse()
}

type serviceA struct {
   repo1 Repository1
   repo2 Repository2
   txMgr TransactionManager
}

func NewServiceA(repo1 Repository1, repo2 Repository2, txMgr TransactionManager) ServiceA {
  return &serviceA{
    repo1: repo1,
    repo2: repo2,
    txMgr: txMgr,
   }
}
```

IMO, this is enough to start developing and testing the application.

**Comment by xcoulon** [comment]
What I'm trying to say is: let's start building sandbox-auth with a basic design as I suggested above and let's add abstraction layers only when needed, instead of building an application framework at first.

**Pull request merged**